tests/contrib/httpx/test_httpx.py

```diff
@@ -0,0 +1,315 @@
+        assert resp.status_code == 500
+
+    with snapshot_context():
+        async with httpx.AsyncClient() as client:
+            resp = await client.get(url)
+            assert resp.status_code == 500
+
+
+@pytest.mark.asyncio
+async def test_split_by_domain(snapshot_context):
+    """
+    When split_by_domain is configure
+        We set the service name to the <host>:<port>
+    """
+    url = get_url("/status/200")
+
+    with override_config("httpx", {"split_by_domain": True}):
+        with snapshot_context():
+            resp = httpx.get(url)
+            assert resp.status_code == 200
+
+        with snapshot_context():
+            async with httpx.AsyncClient() as client:
+                resp = await client.get(url)
+                assert resp.status_code == 200
+
+
+@pytest.mark.asyncio
+async def test_trace_query_string(snapshot_context):
+    """
+    When trace_query_string is enabled
+        We include the query string as a tag on the span
+    """
+    url = get_url("/status/200?some=query&string=args")
+
+    with override_http_config("httpx", {"trace_query_string": True}):
+        with snapshot_context():
+            resp = httpx.get(url)
+            assert resp.status_code == 200
+
+        with snapshot_context():
+            async with httpx.AsyncClient() as client:
+                resp = await client.get(url)
+                assert resp.status_code == 200
+
+
+@pytest.mark.asyncio
+async def test_request_headers(snapshot_context):
+    """
+    When request headers are configured for this integration
+        We add the request headers as tags on the span
+    """
+    url = get_url("/response-headers?Some-Response-Header=Response-Value")
+
+    headers = {
+        "Some-Request-Header": "Request-Value",
+    }
+
+    try:
+        config.httpx.http.trace_headers(["Some-Request-Header", "Some-Response-Header"])
+        with snapshot_context():
+            resp = httpx.get(url, headers=headers)
+            assert resp.status_code == 200
+
+        with snapshot_context():
+            async with httpx.AsyncClient() as client:
+                resp = await client.get(url, headers=headers)
+                assert resp.status_code == 200
+    finally:
+        config.httpx.http = HttpConfig()
+
+
+@pytest.mark.asyncio
+async def test_distributed_tracing_headers():
+    """
+    By default
+        Distributed tracing headers are added to outbound requests
+    """
+    url = get_url("/headers")
+
+    def assert_request_headers(response):
+        data = response.json()
+        assert "X-Datadog-Trace-Id" in data["headers"]
+        assert "X-Datadog-Parent-Id" in data["headers"]
+        assert "X-Datadog-Sampling-Priority" in data["headers"]
+
+    resp = httpx.get(url)
+    assert_request_headers(resp)
+
+    async with httpx.AsyncClient() as client:
+        resp = await client.get(url)
+        assert_request_headers(resp)
+
+
+@pytest.mark.asyncio
+async def test_distributed_tracing_disabled():
```

Comment:
Is distributed tracing a new feature? Is it currently supported in most of our integrations?

Reply:
distributed tracing has been around since the beginning.

it is supported in two ways in integrations:

1. Integrations which make calls to third party applications, these are typically "client" libraries (like `httpx`, `requests`, `grpc` client, etc).

For these integrations we want to "inject" our distributed tracing metadata into the requests being made so they are available by the downstream application to access.

2. Web server integrations (e.g. `flask`, `django`, etc) who will try to "extract" the distributed tracing metadata from the inbound requests to properly associate it's spans with the upstream services metadata.


This is how we get connected distributed traces (e.g. Service A -> Service B -> Service C all in the same trace/flamegraph).